projects/vdk-plugins/vdk-data-sources/README.md

```diff
@@ -0,0 +1,91 @@
+# data-sources
+
+Enables Versatile Data Kit (VDK) to integrate with various data sources by providing a unified interface for data ingestion and management.
+
+The data-sources project is a plugin for the Versatile Data Kit (VDK). It aims to simplify data ingestion from multiple sources by offering a single, unified API. Whether you're dealing with databases, REST APIs, or other forms of data, this project allows you to manage them all in a consistent manner. This is crucial for building scalable and maintainable data pipelines.
+
+
+## Usage
+
+```
+pip install vdk-data-sources
+```
+
+### Concepts
+
+
+#### Data Source
+A Data Source is a central component responsible for establishing and managing a connection to a specific set of data. It interacts with a given configuration and maintains a stateful relationship with the data it accesses. This stateful relationship can include information such as authentication tokens, data markers, or any other form of metadata that helps manage the data connection. The Data Source exposes various data streams through which data can be read.
+
+#### Data Source Stream
+A Data Source Stream is an abstraction over a subset of data in the Data Source. It can be thought of as a channel through which data flows. Each Data Source Stream has a unique name to identify it and includes methods to read data from the stream. For example for Database based data source , each table could be a separate stream. Streams can be ingested in parallel potentially.
+
+Reading from the stream yields a sequence of Data Source Payloads
```

Comment:
VDK is not really a streaming platform. 

But what could happen depends on how the Kafka Data source is implemented 
 You can implement it where it would fetch all data until start_timestamp . In this case it should end. 
 
 But one can reuse https://pypi.org/project/pipelinewise-tap-kafka/ with vdk-singer . 

 And the way they have handled it is to use max_runtime_ms (The maximum time for the tap to collect new messages from Kafka topic) to end the ingestion batch.